### ISSUE TITLE
Curious about CSPN++ baseline

### ISSUE THREAD

**Issue by longyangqi** [XinJCheng/CSPN#31]
Thanks for the awesome paper CSPN++ !
I'm curious about the results of the baseline  "Ma, Cavalheiro, and Karaman 2019".
On kitti depth completion validation,  your implementation gets RMSE 799.08, compare to RMSE 856.75 of the original paper.
Is there any difference between your implementation and the original paper?

Thanks !

**Comment by JUGGHM** [comment]
> Thanks for the awesome paper CSPN++ !
> I'm curious about the results of the baseline  "Ma, Cavalheiro, and Karaman 2019".
> On kitti depth completion validation,  your implementation gets RMSE 799.08, compare to RMSE 856.75 of the original paper.
> Is there any difference between your implementation and the original paper?
> Thanks !

I didn't find the result in either of the papers or Readme.md. Would you like to share the source of an RMSE of 799.08 on the validation set, or is it your experiment result? Thanks!

**Comment by JUGGHM** [comment]
> Thanks for the awesome paper CSPN++ !
> I'm curious about the results of the baseline  "Ma, Cavalheiro, and Karaman 2019".
> On kitti depth completion validation,  your implementation gets RMSE 799.08, compare to RMSE 856.75 of the original paper.
> Is there any difference between your implementation and the original paper?
> Thanks !

If it is your experiment results, I guess it is the mirror connected upProj layer that matters.

**Comment by longyangqi** [comment]
> > Thanks for the awesome paper CSPN++ !
> > I'm curious about the results of the baseline  "Ma, Cavalheiro, and Karaman 2019".
> > On kitti depth completion validation,  your implementation gets RMSE 799.08, compare to RMSE 856.75 of the original paper.
> > Is there any difference between your implementation and the original paper?
> > Thanks !
> 
> If it is your experiment results, I guess it is the mirror connected upProj layer that matters.

You can find the result in paper [[CSPN++](https://arxiv.org/abs/1911.05377)] (Table 1).
In this paper, the author didn't mention the upProj layer.
I also wonder if there is any different configuration compared to the original paper of Ma.

**Comment by JUGGHM** [comment]
> Thanks for the awesome paper CSPN++ !
> I'm curious about the results of the baseline  "Ma, Cavalheiro, and Karaman 2019".
> On kitti depth completion validation,  your implementation gets RMSE 799.08, compare to RMSE 856.75 of the original paper.
> Is there any difference between your implementation and the original paper?
> Thanks !
> 
> If it is your experiment results, I guess it is the mirror connected upProj layer that matters.
> 
> You can find the result in paper [CSPN++] (Table 1).
> In this paper, the author didn't mention the upProj layer.
> I also wonder if there is any different configuration compared to the original paper of Ma.

With the code of Ma's I got a result of 856 on the validation set with a pretrained resnet18 encoder and default configurations, 20 higher than that in Ma's paper. I suspect that a resnet34 encoder will lead to a better result. If you have done such experiment, I would be grateful if you can share the configurations and results.

**Comment by longyangqi** [comment]
> With the code of Ma's I got a result of 856 on the validation set with a pretrained resnet18 encoder and default configurations, 20 higher than that in Ma's paper. I suspect that a resnet34 encoder will lead to a better result. If you have done such experiment, I would be grateful if you can share the configurations and results.

Due to the limited GPU, I only trained the model by cropping the image to [256, 512].
Using a pretrained resnet34 encoder, I got a result of RMSE 839 on validation set (epoch 10 is the best of all 16 epochs, I should train more epochs..., maybe I think 839 is good enough at that time...).
My configurations: lr set to 0.001 with a decay of 0.5 every 10 epochs， batch size set to 4 with 1*1080Ti. 
Now I am trying to train the model on the original image size with a smaller lr.

**Comment by JUGGHM** [comment]
> With the code of Ma's I got a result of 856 on the validation set with a pretrained resnet18 encoder and default configurations, 20 higher than that in Ma's paper. I suspect that a resnet34 encoder will lead to a better result. If you have done such experiment, I would be grateful if you can share the configurations and results.
> 
> Due to the limited GPU, I only trained the model by cropping the image to [256, 512].
> Using a pretrained resnet34 encoder, I got a result of RMSE 839 on validation set (epoch 10 is the best of all 16 epochs, I should train more epochs..., maybe I think 839 is good enough at that time...).
> My configurations: lr set to 0.001 with a decay of 0.5 every 10 epochs， batch size set to 4 with 1*1080Ti.
> Now I am trying to train the model on the original image size with a smaller lr.

That's a good result. I will validate resnet34 encoding 1216x352 images during training if time and GPU are abundant. But I think the baseline of Ma's is kind of large (About 8G?). And another small question: did you crop the images by bottomcrop with Ma's code?

**Comment by longyangqi** [comment]
Yes. First bottomcrop to 1216x256, then random crop to 512x256.

**Comment by XinJCheng** [comment]
Thanks for your attention, Maybe I can share some experiences with this.
1. Crop the image size to 512x256 will reduce the accuracy. 
2. During training, we use the official 1K validation images as our validation set and merge the remain 6K images to our training set
For training codes, we use the codes which Ma open-sourced in GitHub.
If you have any questions, feel free to contact me.

**Comment by XinJCheng** [comment]
> Yes. First bottomcrop to 1216x256, then random crop to 512x256.

Thanks for your attention, Maybe I can share some experiences with this.

Crop the image size to 512x256 will reduce the accuracy.
During training, we use the official 1K validation images as our validation set and merge the remain 6K images to our training set
For training codes, we use the codes which Ma open-sourced in GitHub.
If you have any questions, feel free to contact me.

**Comment by longyangqi** [comment]
> > Yes. First bottomcrop to 1216x256, then random crop to 512x256.
> 
> Thanks for your attention, Maybe I can share some experiences with this.
> 
> Crop the image size to 512x256 will reduce the accuracy.
> During training, we use the official 1K validation images as our validation set and merge the remain 6K images to our training set
> For training codes, we use the codes which Ma open-sourced in GitHub.
> If you have any questions, feel free to contact me.

Thanks for sharing the experiences!
